src/targets/gpu/kernels/include/migraphx/kernels/index.hpp

```diff
@@ -36,44 +36,33 @@ namespace migraphx {
 #if defined(MIGRAPHX_NGLOBAL) && defined(MIGRAPHX_NLOCAL)
 #define MIGRAPHX_NGROUP ((MIGRAPHX_NGLOBAL + MIGRAPHX_NLOCAL - 1) / MIGRAPHX_NLOCAL)
 #endif
+inline __device__ __attribute__((const)) index_int compute_max_local_size()
+{
+#ifdef MIGRAPHX_NLOCAL
+    return MIGRAPHX_NLOCAL;
+#else
+    MIGRAPHX_ASSERT(compute_global_size() % blockDim.x == 0);
+    return blockDim.x;
```

Comment:
So Brian Sumner mentioned, we could call `__ockl_get_enqueued_local_size(0)` to get the uniform size instead removing the need for the assert.

Reply:
`__ockl_get_enqueued_local_size(0)` gets classified as reserved since it starts with `__`